src/datasources/asset/data-sources/calibration-forecast/CalibrationForecastDataSource.ts

```diff
@@ -130,36 +172,12 @@ export class CalibrationForecastDataSource extends AssetDataSourceBase {
                     if (!options.replaceVariables) {
                         return url;
                     }
-
                     const value = options.replaceVariables(`\${__data.fields.${timeGrouping}}`);
```

Comment:
We could have something like this if we don't have a value for the map every time.

```
                    const value = options.replaceVariables(`\${__data.fields.${timeGrouping}}`);
                    const valueAsDate = originalValuesMap.get(value);
                    if (!value && !valueAsDate) {
                        return url;
                    }

                    return `${url}&from=${valueAsDate.from}&to=${valueAsDate.to}`;
```

If we do have a value every time, what about something like this?

```
                    const value = options.replaceVariables(`\${__data.fields.${timeGrouping}}`);
                    if (!value) {
                        return url;
                    }

                    const [from, to] = originalValuesMap.get(value)!;
                    return `${url}&from=${from}&to=${to}`;
```